llvm/lib/Support/Windows/Path.inc

```diff
@@ -785,9 +785,20 @@ std::error_code status(const Twine &path, file_status &result, bool Follow) {
 
   DWORD Flags = FILE_FLAG_BACKUP_SEMANTICS;
   if (!Follow) {
-    DWORD attr = ::GetFileAttributesW(path_utf16.begin());
-    if (attr == INVALID_FILE_ATTRIBUTES)
-      return getStatus(INVALID_HANDLE_VALUE, result);
+    DWORD attr;
+
+    // If getting file attributes fails due to a pending deletion, try
+    // again in a loop to avoid returning a misleading permission denied
+    // error.
+    for (int Retry = 200; Retry >= 0; --Retry) {
+      attr = ::GetFileAttributesW(path_utf16.begin());
+      if (attr != INVALID_FILE_ATTRIBUTES)
+        break;
+      std::error_code code = getStatus(INVALID_HANDLE_VALUE, result);
+      if (code != llvm::errc::delete_pending || !Retry)
+        return code;
+      ::Sleep(10);
```

Comment:
Why the 10s delay? Doesn't that seem too long?

Reply:
void Sleep(
  [in] DWORD dwMilliseconds
);

Reply:
Gah, too many different sleeps with different durations 😮‍💨. Thanks for the reminder about the units!

Reply:
I think that we should bump this to 15ms. IIRC that is the new scheduling quanta.

Reply:
Sure, happy to move it to 15ms. I picked 10ms because that's what's used in [`rename`](https://github.com/llvm/llvm-project/blob/a758e46088b18e49d5aa208e4ed234f6a2bba8e5/llvm/lib/Support/Windows/Path.inc#L591).